test/micro/org/openjdk/bench/java/lang/foreign/SegmentBulkCopy.java

```diff
@@ -98,20 +102,27 @@ enum Alignment {ALIGNED, UNALIGNED}
 
         @Setup
         public void setup() {
-            srcArray = new byte[size + 1];
-            dstArray = new byte[size + 1];
+            // A long array is likely to be aligned at 8-byte boundaries
+            long[] baseArray;
+
+            baseArray = new long[size / Long.BYTES + 1];
+            var rnd = new Random(42);
+            for (int i = 0; i < baseArray.length; i++) {
+                baseArray[i] = rnd.nextLong();
+            }
 
-            switch (Segment.SegmentType.valueOf(segmentType)) {
+            switch (SegmentType.valueOf(segmentType)) {
                 case HEAP -> {
-                    srcSegment = MemorySegment.ofArray(srcArray);
-                    dstSegment = MemorySegment.ofArray(dstArray);
+                    srcSegment = MemorySegment.ofArray(baseArray);
+                    dstSegment = MemorySegment.ofArray(baseArray.clone());
                 }
                 case NATIVE -> {
-                    srcSegment = Arena.ofAuto().allocate(srcArray.length, Long.BYTES);
-                    dstSegment = Arena.ofAuto().allocate(dstArray.length, Long.BYTES);
+                    var s = MemorySegment.ofArray(baseArray);
+                    srcSegment = Arena.ofAuto().allocateFrom(JAVA_LONG, s, JAVA_LONG, 0L, baseArray.length);
+                    dstSegment = Arena.ofAuto().allocateFrom(JAVA_LONG, s, JAVA_LONG, 0L, baseArray.length);
```

Comment:
You can use the other overload that accepts a long[]:

```suggestion
                    srcSegment = Arena.ofAuto().allocateFrom(JAVA_LONG, baseArray);
                    dstSegment = Arena.ofAuto().allocateFrom(JAVA_LONG, baseArray);
```